conan/tools/cmake/toolchain/blocks.py

```diff
@@ -278,11 +281,16 @@ def context(self):
             raise ConanException('CMakeToolchain needs tools.android:ndk_path configuration defined')
         android_ndk_path = android_ndk_path.replace("\\", "/")
 
+        use_cmake_legacy_toolchain = self._conanfile.conf.get("tools.android:cmake_legacy_toolchain")
+        legacy_toolchain_flag_defined = use_cmake_legacy_toolchain is not None
+
         ctxt_toolchain = {
             'android_platform': 'android-' + str(self._conanfile.settings.os.api_level),
             'android_abi': android_abi(self._conanfile),
             'android_stl': libcxx_str,
             'android_ndk_path': android_ndk_path,
+            'android_define_legacy_toolchain_flag': legacy_toolchain_flag_defined,
```

Comment:
It would be a bit simpler to use just 1 variable, if it is defined, it contains the value, if not, nothing to add.